src/components/sketches/test4.ts

```diff
@@ -0,0 +1,110 @@
+import p5 from 'p5';
+export default function sketch(p: p5): void {
+  let canvas: p5.Renderer;
+
+  const particles: Particle[] = [];
+  const windowWidth = window.innerWidth;
+  const windowHeight = window.innerHeight;
+
+  p.setup = () => {
+    canvas = p.createCanvas(window.innerWidth, window.innerHeight);
+    canvas.id('p5-background');
+
+    const particlesLength = Math.min(Math.floor(window.innerWidth / 10), 200); //note this line
+    for (let i = 0; i < particlesLength; i++) {
+      particles.push(new Particle());
+    }
+  };
+
+  p.windowResized = () => {
+    p.resizeCanvas(window.innerWidth, window.innerHeight, false);
```

Comment:
in this case we should set redraw to true because the position of a Particles are dependent on the window width and height

Reply:
Gotcha